### PR TITLE
Restore batching memory control functionality to LightningGPU

### PR DESCRIPTION
**Pull request by mlxd**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:** This PR fixes batching support for LightningGPU devices, and adds preliminary support to enable it for LQ and LK at a later date.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPU.hpp [hunk 1]
```diff
@@ -195,11 +195,11 @@ class AdjointJacobian final
...
         /// overwrite the 0 jacs values before returning
-        for (std::size_t i = 0; i < futures.size(); i++) {
+        for (std::size_t i = 0; i < jac_futures.size(); i++) {
             const auto first = static_cast<std::size_t>(
                 std::ceil(obs.size() * i / num_chunks));
 
-            auto jac_chunk = futures[i].get();
+            auto jac_chunk = jac_futures[i].get();
             for (std::size_t j = 0; j < jac_chunk.size(); j++) {
                 std::copy(jac_chunk.begin(), jac_chunk.end(),
```
Comment (multiphaseCFD) [inline pennylane_lightning/core/src/simulators/lightning_gpu/algorithms/AdjointJacobianGPU.hpp:204]: As we would like to use the incoming  jac span? Can we add a //TODO here for it?

File: pennylane_lightning/lightning_gpu/lightning_gpu.py [hunk 2]
```diff
@@ -691,24 +692,69 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
...
+                    num_obs = len(processed_data["obs_serialized"])
+                    batch_size = (
+                        num_obs
+                        if isinstance(self._batch_obs, bool)
+                        else self._batch_obs * self._dp.getTotalDevices()
+                    )
+                    jac = []
+                    for chunk in range(0, num_obs, batch_size):
+                        obs_chunk = processed_data["obs_serialized"][chunk : chunk + batch_size]
+                        jac_chunk = adjoint_jacobian.batched(
```
Comment (multiphaseCFD) [inline pennylane_lightning/lightning_gpu/lightning_gpu.py:706]: I'm wondering if for loop here can serialize the operation or not?

File: pennylane_lightning/core/_serialize.py [hunk 3]
```diff
@@ -52,9 +52,12 @@ class QuantumScriptSerializer:
     """
```
Comment (AmintorDusko) [inline pennylane_lightning/core/_serialize.py:52]: To keep note, the class docstring is incomplete. It was missing use_mpi and now split_obs.

**Review by multiphaseCFD — APPROVED** [review]
Looks good to me! Thanks @mlxd !

**Review by AmintorDusko — APPROVED** [review]
Coverage will need some work. Thanks for your work.

**Pull request merged**